metaflow/plugins/events_decorator.py

```diff
@@ -597,9 +597,41 @@ def format_deploytime_value(self):
             # Entire trigger is a function (returns either string or dict)
             old_trig = trigger
             if isinstance(trigger, DeployTimeField):
+                # convert the trigger to string or dict
                 trigger = deploy_time_eval(trigger)
+                if is_stringish(trigger):
+                    pass
+                elif isinstance(trigger, dict):
+                    if "name" not in trigger:
+                        raise MetaflowException(
+                            "The *flow* attribute for *@trigger_on_finish* is missing the "
+                            "*name* key."
+                        )
+                    flow_name = trigger["name"]
+
+                    if not is_stringish(flow_name) or "." in flow_name:
+                        raise MetaflowException(
+                            "The *name* attribute of the *flow* is not a valid string"
+                        )
+                    result = {"fq_name": flow_name}
+                    if "project" in trigger:
+                        if is_stringish(trigger["project"]):
+                            result["project"] = trigger["project"]
+                        else:
+                            raise MetaflowException(
+                                "The *project* attribute of the *flow* is not a string"
+                            )
+                    if "project_branch" in trigger:
+                        if is_stringish(trigger["project_branch"]):
+                            result["project_branch"] = trigger["project_branch"]
```

Comment:
Can you extract out lines 602-630 into a helper function? The code is identical to lines 397-429. Also on line 425 it's setting result to `branch` instead of `project_branch`. Which one is correct?